fabric-key-binding-api-v1/src/main/java/net/fabricmc/fabric/api/client/keybinding/v1/KeyBindingRegistry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.keybinding.v1;
+
+import net.fabricmc.fabric.impl.client.keybinding.KeyBindingRegistryImpl;
+
+/**
+ * Interface for registering key bindings.
+ *
+ * @see FabricKeyBinding
+ */
+public interface KeyBindingRegistry {
+	KeyBindingRegistry INSTANCE = KeyBindingRegistryImpl.INSTANCE;
+
+	/**
+	 * Add a new key binding category.
+	 *
+	 * @param categoryName The key binding category name.
+	 * @return True if a new category was added.
+	 */
+	boolean addCategory(String categoryName);
```

Comment:
If we add register to the builder's build(), this method could get moved as a static helper to KeyCategories.

Reply:
should we just automatically register the category if it doesn't exist?

Reply:
Seems like we are already doing that with a warning that doesn't make sense, I think I will just remove that warning.

Reply:
Just registering the category makes sense